### DISCUSSION TITLE
Can we work on crowdfunding functionality

### DISCUSSION THREAD

**Discussion by GiverofMemory** [MrNaif2018/bitcart#262] [Ideas]
I love this project and it is the perfect fit for my online business.  But there is an even more pressing need and that is uncensorable crowdfunding.  Can we work on a solution to this?  

Also I raised the issue with the dogecoin devs to make an electrum fork for doge.  Please contribute to the discussion https://github.com/dogecoin/dogecoin/discussions/2831!

**Comment by MrNaif2018** [comment]
Hello! Yes, crowdfunding is planned, any help is welcome! We will need some initial discussion and the implementation can start
Do you know flipstarter by the way? What I wonder is, do we implement crowdfunding via some onchain funds locking like they do afaik or just something like regular invoice generation workflow would suffice?

As for dogecoin, I'll track the discussion, if their devs want to maintain such a fork I will guide them if needed

**Comment by MrNaif2018** [comment]
From our telegram:

> I see https://en.bitcoin.it/wiki/Contract#Example_3:_Assurance_contracts
> And it would work for us, users create a transaction with their wallet to required address with required amount, adding ALL | ANYONECANSPEND sighash, but do not broadcast tx to network but instead give to the server which adds new input to the total transaction
> When enough money is collected the transaction can be just broadcasted as sum of inputs is >= sum of outputs
> 
> But there are a few problems:
> 1) Do many wallets supporting adding ALL | ANYONECANSPEND hash to inputs? Is there a way to do it available?
> 2) As we need to send transaction to the server, I guess broadcasting it wouldn't work. Do all wallets provide ability to copy raw transaction?
> 
> UX is already not that optimal, and we want to try to improve it while staying as secure as possible
> 
> everyone do you have any ideas, especially if you know how crowdfunding contracts like that usually work, keeping the best UX?

We have a way to make it secure like flipstarter, but some UX issues are there
